src/main/scala/tech/cryptonomic/conseil/tezos/TezosRemotes.scala

```diff
@@ -0,0 +1,202 @@
+package tech.cryptonomic.conseil.tezos
+
+import tech.cryptonomic.conseil.config.{HttpStreamingConfiguration, NetworkCallsConfiguration}
+import tech.cryptonomic.conseil.config.Platforms.TezosConfiguration
+import tech.cryptonomic.conseil.generic.chain.RemoteRpc
+import tech.cryptonomic.conseil.util.JsonUtil.JsonString
+
+
+object TezosRemoteInstances {
+
+  object Akka {
+    import com.typesafe.scalalogging.Logger
+    import cats.Monoid
+    import scala.concurrent.Future
+    import akka.actor.ActorSystem
+    import akka.stream.ActorMaterializer
+    import akka.stream.scaladsl.Source
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.model._
+
+    case class RemoteContext(
+      tezosConfig: TezosConfiguration,
+      requestConfig: NetworkCallsConfiguration,
+      streamingConfig: HttpStreamingConfiguration
+    )(implicit val system: ActorSystem) {
+
+      object ShutdownComplete extends ShutdownComplete
+      private val rejectingCalls = new java.util.concurrent.atomic.AtomicBoolean(false)
```

Comment:
Ok, good question, let me know if you have ideas because the concept lived on...

When the system starts shutting down, it might happen that Lorre is still running its batch of requests (tens or hundreds of those).
What happened was that those requests were run pointlessly and returned an akka http stream response that would generate lots of noisy errors about the body not being consumed within the given timeout (If I remember correctly).

So I introduced (time ago) this sort of short-circuiting state that would immediately reject any node request from lorre from when the system initiate its shutdown process (even if fired by an OS signal, like Ctrl-C)